InternalDocs/asyncio.md

```diff
@@ -205,6 +206,119 @@ In free-threading, it avoids contention on a global dictionary as
 threads can access the current task of thier running loop without any
 locking.
 
+---
+
+# async generators
+
+This section describes the implementation details of async generators in `asyncio`.
+
+Since async generators are meant to be used from coroutines,
+the finalization (execution of finally blocks) of it needs
+to be done while the loop is running.
+Most async generators are closed automatically
+when they are fully iterated over and exhausted, however,
+if the async generator is not fully iterated over,
+it may not be closed properly, leading to the `finally` blocks not being executed.
+
+Consider the following code:
+```py
+import asyncio
+
+async def agen():
+    try:
+        yield 1
+    finally:
+        await asyncio.sleep(1)
+        print("finally executed")
+
+
+async def main():
+    async for i in agen():
+        break
+
+loop = asyncio.EventLoop()
+loop.run_until_complete(main())
+```
+
+The above code will not print "finally executed", because the
+async generator `agen` is not fully iterated over
+and it is not closed manually by awaiting `agen.aclose()`.
```

Comment:
Is `agen.aclose()` a shortcut for something? You cannot literally do that (`'function' object has no attribute 'aclose'`). So what should I wrote to close it manually (if I was so inclined)?

Reply:
You can do something like this to close it:
```py
async def main():
    g = agen()
    async for i in g:
        break

    await g.aclose()
```